src/libopensc/opensc.h

```diff
@@ -940,9 +946,11 @@ int sc_cancel(sc_context_t *ctx);
 /**
  * Tries acquire the reader lock.
  * @param  card  The card to lock
+ * @param  resetok If nonzero means to continue even if the card was reset
+ *	as long as it wasn't reset between individual operations
  * @retval SC_SUCCESS on success
  */
-int sc_lock(struct sc_card *card);
+int sc_lock(struct sc_card *card, int resetok);
```

Comment:
> If I saw correctly, sc_lock() is never called with resetok == 1; why do we need this then?

This is currently unused but it can be used in a future code so the card isn't reinitialized
needlessly where we don't care if it was reset in meantime.
It looks like supporting cards with multiple applets would require reloading the context before
every transaction so there could be also other sources of resets than minidriver
deauthentication (as changing the current applet is essentially equivalent to a reset).

> An alternative approach would be to set a flag in the card if a reset has happened.
> If so, the card driver may re-initialize the card if needed. 

We do set a flag in an non-minidriver case (commit c21086648f01) where we get this information from PC/SC.
But in the minidriver case this does not work (we don't get this information from PC/SC) so we need to keep track ourselves that the reset happened (because the card was deauthenticated).
When it happens a flag is set ("needs_reinit"), just inside minidriver, not PC/SC driver.